### DISCUSSION TITLE
有考虑为cache增加一个清除的信号或者API么？

### DISCUSSION THREAD

**Discussion by ihipop** [IrineSistiana/mosdns#549] [Q&A] [answered]
提出这个问题的前提是**如果provider加载的规则如果数据库文件有更新，应该不会invalid 所有cache吧？**
这样就存在一个尴尬的问题：我的规则文件更新了，但是因为cache没失效，导致我编写的流程（IPSET维护）得等到cache自己过期后才能触发或者重启程序。

如果上述问题不存在，此问题close,否则是否有必要做一个cache invalid但是不需要重启程序的方法？
比如：
通过`kill -INT` 或者API请求？

**Comment by urlesistiana** [comment, marked as answer]
> 如果provider加载的规则如果数据库文件有更新，应该不会invalid 所有cache吧

不会。

会考虑用 api 实现。
